libs/NfcCoreLib/lib/Fri/phFriNfc_ISO15693Map.c

```diff
@@ -1586,15 +1629,52 @@ phFriNfc_ReadRemainingInMultiple (
 {
     NFCSTATUS result = NFCSTATUS_FAILED;
     phFriNfc_ISO15693Cont_t *ps_iso_15693_con = &(psNdefMap->ISO15693Container);
+    phHal_sIso15693Info_t   *ps_iso_15693_info =
+                            &(psNdefMap->psRemoteDevInfo->RemoteDevInfo.Iso15693_Info);
     uint32_t remaining_size = ISO15693_GET_REMAINING_SIZE(ps_iso_15693_con->max_data_size,
                                            startBlock, 0);
     // Check capabilities
     if (ps_iso_15693_con->read_capabilities & ISO15693_CC_USE_MBR) {
         // Multi-page read command
-        uint8_t mbread[1];
-        mbread[0] = (uint8_t)((remaining_size / ISO15693_BYTES_PER_BLOCK) - 1);
-        result = phFriNfc_ISO15693_H_ReadWrite (psNdefMap, ISO15693_READ_MULTIPLE_COMMAND,
-                mbread, 1);
+        uint8_t mbread[2];
+        uint8_t mbread_len = 1;
+        uint32_t nb_blocks = 0;
+        uint8_t command = ISO15693_READ_MULTIPLE_COMMAND;
+
+        /* Compute how many block can be read at a time.
+           If we are in NCI2.0 mode, MaxNFCVFrameSize is set and we need to split the packet to MaxNFCVFrameSize - 1.
+           In case of M24LR tag, we can read 32 blocks maximum if they are all located in the same sector.
+         */
+        nb_blocks = ((remaining_size / ISO15693_BYTES_PER_BLOCK) - 1);
+        if (ISO15693_PROTOEXT_FLAG_REQUIRED(ps_iso_15693_info->Uid))
+        {
+            if ((nb_blocks + (ps_iso_15693_con->current_block % ISO15693_STM_M24LR_MAX_BLOCKS_READ_PER_SECTOR)) >= ISO15693_STM_M24LR_MAX_BLOCKS_READ_PER_SECTOR - 1)
+            {
+                nb_blocks = ISO15693_STM_M24LR_MAX_BLOCKS_READ_PER_SECTOR - (ps_iso_15693_con->current_block % ISO15693_STM_M24LR_MAX_BLOCKS_READ_PER_SECTOR) - 1;
+            }
+            mbread_len = 2;
```

Comment:
Shoudn't this be followed by `command = ISO15693_EXT_READ_MULTIPLE_COMMAND;` ?

Reply:
In this block ISO15693_PROTOEXT_FLAG_REQUIRED means that we are using tags not 100% inline with the standard where ISO15693_READ_MULITPLE_COMMAND takes 2 bytes for block number instead of 1.